### PR TITLE
remove some subquery and join constraints

### PR DESCRIPTION
**Pull request by brunocalza**
# Summary

This PR removes the following constraints
- subquery on inserts
- subquery on update
- subquery on deletes
- subquery and joins inside a insert + select statement

cc @dtbuchholz @asutula 

# Context

https://linear.app/tableland/issue/ENG-877/allow-more-complex-sql-subqueries-in-parser

### CODE REVIEW

File: parser_test.go [hunk 1]
```diff
@@ -4773,63 +4773,6 @@ func TestLimits(t *testing.T) {
 }
 
 func TestDisallowSubqueriesOnStatements(t *testing.T) {
```
Comment (dtbuchholz) [inline parser_test.go:4775]: instead of removing these tests, would we want to just validate the statements _do_ work?

**Review by dtbuchholz — APPROVED** [review]
lgtm! left a comment but feel free to ignore if it doesn't make sense...those can be done at the [JS client](https://github.com/tablelandnetwork/go-sqlparser/blob/main/js/test/main.test.cjs), if needed

File: parser_test.go (hunk 1 above)
Comment (brunocalza) [reply parser_test.go:4775]: Yeah sure

File: parser_test.go [hunk 2]
```diff
@@ -4772,99 +4772,39 @@ func TestLimits(t *testing.T) {
 	})
 }
 
-func TestDisallowSubqueriesOnStatements(t *testing.T) {
+func TestAllowSubqueriesOnStatements(t *testing.T) {
```
Comment (brunocalza) [inline parser_test.go:4775]: cc @dtbuchholz

**Review by asutula — APPROVED** [review]
Great! Thanks for getting this done.

**Pull request merged**